shared/gh/api/gh.api.util.js

```diff
@@ -50,20 +61,34 @@ define(['exports', 'moment', 'sinon', 'bootstrap-notify'], function(exports, mom
      * Get the date range the calendar should be displaying. The range starts on January 1st of the
      * previous year and ends December 31st of the next year.
      *
-     * @return {Object}     Object containg start and end date that form the range of the calendar
+     * @param  {Function}   callback            Standard callback function
+     * @param  {Object}     callback.range      Object containg start and end date that form the range of the calendar
      */
-    var getCalendarDateRange = exports.getCalendarDateRange = function() {
-        // Get the current year to base calculations off of
-        var currentYear = new Date().getFullYear();
+    /* istanbul ignore next */
+    var getCalendarDateRange = exports.getCalendarDateRange = function(callback) {
 
         // Create the range object to return
         var range = {
-            'start': (currentYear - 1) + '-01-01',
-            'end': (currentYear + 1) + '-12-31'
+            'start': null,
+            'end': null
         };
 
-        // Return the range object
-        return range;
+        // Fetch the calendar's current view type
+        $(document).trigger('gh.calendar.getCurrentView', function(currentView) {
```

Comment:
I'm not sure why we'd call a function that then sends out an event with the data we want?